iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,343 @@
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSenderIsSuccessful)
+{
+    const auto senderPort = m_runtime->getMiddlewareSender(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), senderPort->m_serviceDescription);
+    EXPECT_EQ(22u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_DefaultArgs)
+{
+    const auto senderPort = m_runtime->getMiddlewareSender(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_SenderlistOverflow)
+{
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER; ++i)
+    {
+        m_runtime->getMiddlewareSender(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiverIsSuccessful)
+{
+    auto receiverPort = m_runtime->getMiddlewareReceiver(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), receiverPort->m_serviceDescription);
+    EXPECT_EQ(22u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_DefaultArgs)
+{
+    auto receiverPort = m_runtime->getMiddlewareReceiver(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_ReceiverlistOverflow)
+{
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER + 1; ++i)
+    {
+        m_runtime->getMiddlewareReceiver(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetServiceRegistryChangeCounter_OfferStopOfferService)
+{
+    auto initialValue = m_runtime->getServiceRegistryChangeCounter();
+    auto cout = initialValue->load();
+
+    m_runtime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    auto counter = m_runtime->getServiceRegistryChangeCounter();
```

Comment:
Logically this is correct (I still would have liked EXPECT_GT more...but ok). We need to try to find variable names that convey the meaning of what is happening. I.e. initialCount. It is about comprehensibility for other developers:-)

Reply:
using EXPECT_GT after offerService()  is fine, but using EXPECT_GT again after stopOfferService() may not fit, because value is already incremented for 1st time(during offerService )
It will be like performing EXPECT_GT on value that we have already confirmed to be greater